### PR TITLE
maint: refresh pre-commit, flake8, gitignore config

### PR DESCRIPTION
**Pull request by minrk**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -8,44 +8,56 @@
...
-          - --target-version=py310
+          - --in-place
 
   # Autoformat: Python code
   - repo: https://github.com/pycqa/isort
     rev: 5.11.1
     hooks:
       - id: isort
-        args:
-          - --profile=black
```
Comment (consideRatio) [inline .pre-commit-config.yaml:36]: The flags removed here should be set in a pyproject.toml or similar before merge if removed here I think

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -8,44 +8,56 @@
...
+  autoupdate_interval: monthly
 repos:
   # Autoformat: Python code, syntax patterns are modernized
   - repo: https://github.com/asottile/pyupgrade
     rev: v3.3.1
     hooks:
       - id: pyupgrade
         args:
-          - --py38-plus
+          - --py37-plus
```
Comment (consideRatio) [inline .pre-commit-config.yaml:20]: This project is at py38+

https://github.com/jupyterhub/docker-image-cleaner/blob/main/setup.py#L18

**Review by consideRatio — APPROVED** [review]
LGTM - ready for a merge?

**Comment by minrk** [comment]
@consideRatio sorry, sloppy import from other projects. Should be better now.

**Pull request merged**